lib/elixir/lib/option_parser.ex

```diff
@@ -137,6 +170,32 @@ defmodule OptionParser do
     do_parse(argv, compile_config(opts), [], [], [], false)
   end
 
+  @doc """
+  The same as `parse_head/2` but raises an `ArgumentError` if any invalid
+  options were encountered.
+
+  If there weren't any errors, returns a three-element tuple as follows:
+
+      1. parsed options,
+      2. remaining arguments,
+      3. empty list.
+
+  ## Examples
+
+      iex> OptionParser.parse_head!(["--number", "lib", "test/enum_test.exs", "--verbose"], strict: [number: :integer])
+      ** (ArgumentError) You have 1 error! The option --number exists but it is of the wrong type: "lib"
+
+      iex> OptionParser.parse_head!(["--verbose", "--source", "lib", "test/enum_test.exs", "--unlock"], strict: [verbose: :integer, source: :integer])
+      ** (ArgumentError) You have 2 errors! The option --verbose does not exist. The option --source exists but it is of the wrong type: "lib"
```

Comment:
This message is confusing, we do have `--verbose` but error says it does not exist.